### PR TITLE
libnetwork/iptables: Fix test panic when execute only one test

### LINKED ISSUE #42696: panic in TestConcurrencyWithWait
**Description**
TestConcurrencyWithWait has panic

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x5a383a]

goroutine 16 [running]:
github.com/docker/docker/libnetwork/iptables.(*ChainInfo).Forward(0x0, 0x60c248, 0x2, 0xc00001c390, 0x10, 0x10, 0x4d2, 0x60c3d5, 0x3, 0x60d950, ...)
        /go/src/github.com/docker/docker/libnetwork/iptables/iptables.go:297 +0x3a
github.com/docker/docker/libnetwork/iptables.RunConcurrencyTest.func2(0x0, 0x0)
        /go/src/github.com/docker/docker/libnetwork/iptables/iptables_test.go:213 +0xba
github.com/docker/docker/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000022b40, 0xc000012500)
        /go/src/github.com/docker/docker/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x59
created by github.com/docker/docker/vendor/golang.org/x/sync/errgroup.(*Group).Go
        /go/src/github.com/docker/docker/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
```
**Steps to reproduce the issue:**
```
go test -run TestConcurrencyWithWait github.com/docker/docker/libnetwork/iptables
```
**Describe the results you received:**
panic

**Describe the results you expected:**
test should pass

**Additional information you deem important (e.g. issue happens only occasionally):**

go version go1.16.5 linux/amd64

### PR DESCRIPTION
**Pull request by ch33hau**
fixes #42696

**- What I did**
The test paniced when executing individually is because they are sharing some same variables (`natChain`, `filterChain`, and `bridgeName`), these variables only initiated on the first test. 

**- How I did it**
This PR moves the shared variables to local variables and initiates them within every test.

**- How to verify it**
Run a single test:
```
go test -run TestConcurrencyWithWait github.com/docker/docker/libnetwork/iptables
```
**- Description for the changelog**
Fix iptables tests

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thank you for contributing! It appears your commit message is missing a DCO sign-off,
causing the DCO check to fail.

We require all commit messages to have a `Signed-off-by` line with your name
and e-mail (see ["Sign your work"](https://github.com/moby/moby/blob/v20.10.11/CONTRIBUTING.md#sign-your-work) in the CONTRIBUTING.md in this repository),
which looks something like:

```
Signed-off-by: YourFirsName YourLastName <yourname@example.org>
```

There is no need to open a new pull request, but to fix this (and make CI pass),
you need to _amend_ the commit(s) in this pull request, and "force push" the amended
commit.

You can find some instructions in the output of the DCO check (which can be found
in the "checks" tab on this pull request), as well as in the [Moby contributing guide](https://github.com/moby/moby/blob/v20.10.11/docs/contributing/set-up-git.md).

Steps to do so "roughly" come down to:

1. Set your name and e-mail in git's configuration:

    ```bash
    git config --global user.name "YourFirstName YourLastName"
    git config --global user.email "yourname@example.org"
    ```

    (Make sure to use your _real_ name (**not your GitHub username/handle**) and e-mail)


2. Clone your fork locally
3. Check out the branch associated with this pull request
4. Sign-off and amend the existing commit(s)

    ```bash
    git commit --amend --no-edit --signoff
    ```

    If your pull request contains multiple commits, either squash the commits (if
    needed) or sign-off each individual commit.

5. _Force push_ your branch to GitHub (using the `--force` or [`--force-with-lease`](https://stackoverflow.com/questions/52823692/git-push-force-with-lease-vs-force) flags) to update the pull request.


Sorry for the hassle;  let me know if you need help or more detailed instructions!

**Comment by ch33hau** [comment]
Hi @thaJeztah sorry for that, I updated.

File: libnetwork/iptables/iptables_test.go [hunk 1]
```diff
@@ -196,6 +210,21 @@ func TestConcurrencyNoWait(t *testing.T) {
...
+
+	err = iptable.ProgramChain(natChain, bridgeName, false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = iptable.ProgramChain(filterChain, bridgeName, false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (corhere) [inline libnetwork/iptables/iptables_test.go:226]: How come the chains have to be programmed a second time? Why aren't the `ProgramChain` calls inside `createNewChain` sufficient?

**Review by corhere — COMMENTED** [review]
Thank you so much for fixing up these tests! I just have one question before approving.

File: libnetwork/iptables/iptables_test.go (hunk 1 above)
Comment (ch33hau) [reply libnetwork/iptables/iptables_test.go]: Hi @corhere , thanks for reviewing! 

Thanks for catching this, I can't see any reasons why they are needed, it should be my mistake. I have removed them.

**Review by corhere — APPROVED** [review]
LGTM, thanks!

File: libnetwork/iptables/iptables_test.go [hunk 2]
```diff
@@ -141,12 +147,16 @@ func TestLink(t *testing.T) {
 }
 
 func TestPrerouting(t *testing.T) {
+	iptable, natChain, _, err := createNewChain()
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (thaJeztah) [inline libnetwork/iptables/iptables_test.go:153]: Perhaps we can make `createNewChain()` a `t.Helper()` then we can skip all the "per test" error handling.

**Review by thaJeztah — COMMENTED** [review]
Left one suggestion; I was writing along, and opened a PR against your branch with my suggestions; https://github.com/ch33hau/moby/pull/1

File: libnetwork/iptables/iptables_test.go (hunk 2 above)
Comment (ch33hau) [reply libnetwork/iptables/iptables_test.go]: Thanks @thaJeztah , that looks a lot cleaner in the code!

**Review by thaJeztah — APPROVED** [review]
LGTM!

Thanks so much (was about to ask if you needed help with squashing, but I see you managed to do so ❤️)

**Pull request merged**